src/controllers/mock_controller.rs

```diff
@@ -81,43 +21,62 @@ pub fn mock(stream: TcpStream, http_fields: HttpFields) {
     // let path_formatted = re.replace_all(path, "/register/:id");
     //println!("{:#}", re.is_match(Some(data).unwrap().to_string().as_str()));
 
-    let ret = mock_service::execute(
+    let ret = match_mock::execute(
         Http {
             path: http_path,
             method: http_method,
             request_body: http_fields.body,
+            query_params,
+            headers: http_fields.headers,
         },
-        File {
+        MockFile {
             file_path: String::new(),
         },
     );
 
-    if ret["$.body"]["path"] != Null {
-        status = "HTTP/1.1 404 NOT FOUND";
-    }
-
-    if ret["$.body"]["request"] != Null {
-        status = "HTTP/1.1 400 BAD REQUEST";
+    match ret {
+        TypeOr::Left(ret) => {
+            return stream_response(&stream, ret, String::from("200"));
+        }
+        TypeOr::Right(ret) => {
+            return response(
+                &stream,
+                ret["$.body"].to_owned(),
+                ret["$.status"].to_string(),
+            );
+        }
     }
+}
 
-    return response(
-        &stream,
-        ret["$.body"].to_owned(),
-        status!(ret["$.status"].to_string().as_str(), status),
+fn response_format(status: String, length: String, content_type: String) -> String {
+    let content_header = format!(
+        "Content-Type: {content_type}\r\nAccess-Control-Allow-Origin: *\r\nAccept-Ranges: bytes"
     );
+
+    let status = status!(status.as_str());
+
+    return format!("{status}\r\n{length}\r\n{content_header}\r\n\r\n");
 }
 
-fn response_format(status: String, length: String, content: String) -> String {
-    let content_type: &str = "Content-Type: application/json";
+fn stream_response(mut stream: &TcpStream, data: Vec<u8>, status: String) {
+    let length = format!("Content-Length: {}", data.len());
+    let response = response_format(status, length, "video/mp4".to_string());
```

Comment:
enable to receive as a variable other type of file as `image/jpeg` or `audio/mpeg`...